Update release notes and library guide for oneDPL 2021.6 release (#412)
oneapi-src/oneDPL: documentation/library_guide/parallel_api/buffers_and_usm.rst

    return 0;
  }

.. Note::

>    When using device USM allocations, e.g. via ``malloc_device``, manually copy data to the device
>    before calling oneDPL algorithms and copy them back once the algorithms finished execution.

Reviewer: ```suggestion
   When using device USM, such as allocated by ``malloc_device``, manually copy data to this memory
   before calling oneDPL algorithms, and copy it back once the algorithms have finished execution.
```

>    For more information regarding USM, please refer to SYCL™ 2020 Specification.

Reviewer: Here I would remove the "Note" (i.e. it's just a part of overall description for working with USM, not something very special) and the "For more information" sentence (all needed details about SYCL should be looked there, USM is not special in that regard).
